strapi/strapi
# fix query-builder distinct issue with one-to-many and many-to-many relations

### What does it do?

fixes distinct issues with orderBy in one-to-many and many-to-many relations when we our orderBy creates a join on query
fixes: #13153

### reproduce the issue 

1. on an example project, create two (or more) address entity 
2. then create a category and assign created addresses to it 
3. try to get categories by sorting by a column of addresses for example `/categories?sort=addresses.slug:ASC`
4. you will see that we got two (or more) categories but we just had one ( we'll have duplicates for each address item )

### Why is it needed?

currently, we have an abvious bug in relations query building and its results 
we don't need `order by` columns distinction 
we will  always have a `target.id` select on our entity so all the returned data will be distinct since our id is unique we don't need to distinguish it by relations, distinction by id + relation fields always return duplicate data since all the rows unwinded by the result

summary : 
for relations ordering, we are generating something like this
`select distinct `t2`.`created_at`, `t0`.* from ...` ⛔ 
but we need to distinct by target table to prevent duplicate data 
`select distinct  `t0`.* from ...` 👍 

### How to test it?

1. on an example project, create two (or more) address entity 
2. then create a category and assign created addresses to it 
3. try to get categories by sorting by a column of addresses for example `/categories?sort=addresses.slug:ASC`
4. you will see that we just have one item, not two 👍 

### Related issue(s)/PR(s)

related to #13153
